vivisect/analysis/arm/emulation.py

```diff
@@ -126,6 +126,13 @@ def prehook(self, emu, op, starteip):
             self.logAnomaly(emu, self.fva, "0x%x: (%r) ERROR: %s" % (op.va, op, e))
             logger.warning("0x%x: (%r)  ERROR: %s", op.va, op, e)
 
+    def apicall(self, emu, op, pc, api, argv):
+        logger.info("apiCall(%r, 0x%x, %r, %r)", op, pc, api, argv)
+        norets = emu.vw.getMeta('NoReturnApisVa', {})
+        if norets.get(pc, False):
+            logger.info("=== halting emu: NoReturnApisVa 0x%x", pc)
+            emu.stopEmu()
+        
```

Comment:
makes ArmWorkspaceEmulator honor norets

Reply:
Should this be here? The base emulator has this directly in runFunction, because if we don't have an emulation monitor instance, `apicall` won't be called.

Reply:
i think i understand the concern.  can you point me to the line that does this in the base emulator?  i'm not seeing it.

as for this, this emulation monitor is stamping out new locations into the VW database, so that's the big thing i'm worried about.
currently we don't honor NoReturn in the emulation pass.  at least, not in ARM... possibly not at all.

so perhaps the problem is wider spread and the solution belongs higher up the foodchain.  
let's have this conversation.  i'm responding to specific examples of badness which occur in ARM.  but perhaps we simply haven't noticed this in other architectures.

Reply:
arg.  this is because we use our own ARM-specific `runFunction()`, isn't it.

now that you bring it up, that's more likely the bug i'm solving.

Reply:
fixed and it works now.  good catch!